echopype/echodata/combine.py

```diff
@@ -154,12 +205,250 @@ def check_echodatas_input(echodatas: List[EchoData]) -> Tuple[str, List[str]]:
+        ``channel_selection_dict`` (see below for further details)
+
+    Returns
+    -------
+    dict
+        A dictionary with the same keys as ``has_chan_dim`` and values
+        determined by ``sonar_model`` and ``user_channel_selection`` as follows:
+        - If ``user_channel_selection=None``, then the values of the dictionary
+        will be set to ``None``
+        - If ``user_channel_selection`` is a list, then all keys corresponding to
+        an ``EchoData`` group with a ``channel`` dimension will have their values
+        set to the provided list and all other groups will be set to ``None``
+        - If ``user_channel_selection`` is a dictionary, then all keys corresponding to
+        an ``EchoData`` group without a ``channel`` dimension will have their values
+        set as ``None`` and the other group's values will be set as follows:
+            - If ``sonar_model`` is not EK80-like then all values will be set to
+            the union of the values of ``user_channel_selection``
+            - If ``sonar_model`` is EK80-like then the groups ``Sonar, Platform, Vendor_specific``
+            will be set to the union of the values of ``user_channel_selection`` and the rest of
+            the groups will be set to the same value in ``user_channel_selection`` with the same key
+    """
+
+    # base case where the user did not provide selected channels (will be used downstream)
+    if user_channel_selection is None:
+        return {grp: None for grp in has_chan_dim.keys()}
+
+    # obtain the union of all channels for each beam group
+    if isinstance(user_channel_selection, list):
+        union_beam_chans = user_channel_selection[:]
+    else:
+        union_beam_chans = list(set(itertools.chain.from_iterable(user_channel_selection.values())))
+
+    # make channel_selection dictionary where the keys are the EchoData groups and the
+    # values are based on the user provided input user_channel_selection
+    channel_selection_dict = dict()
+    for ed_group, has_chan in has_chan_dim.items():
+
+        # if there are no channel dimensions in the group, set the value to None
+        if has_chan:
+
+            # if the model is not EK80-like set value to the union of the values
+            # of user_channel_selection
+            if sonar_model in ["EK80", "ES80", "EA640"]:
+
+                # if the model is EK80-like
+                if (ed_group in ["Sonar", "Platform", "Vendor_specific"]) or isinstance(
+                    user_channel_selection, list
+                ):
+                    # set value to the union of the values of user_channel_selection
+                    channel_selection_dict[ed_group] = union_beam_chans
+
+                else:
+
+                    # set value to the user provided input with the same key
+                    channel_selection_dict[ed_group] = user_channel_selection[ed_group]
+
+            else:
+                channel_selection_dict[ed_group] = union_beam_chans
```

Comment:
This logic here seems can be simplified?
The only case when you have `channel_selection_dict[ed_group] = user_channel_selection[ed_group]` is when you are _not_ in the "Sonar", "Platform", "Vendor_specific" groups and when `user_channel_selection` is _not_ a list.
In all other cases `channel_selection_dict[ed_group] = union_beam_chans` is used.

Reply:
I believe I understand where the simplification should occur. I will go ahead and implement it.